Source/RxTest/ThrowError+RxTest.swift

```diff
@@ -0,0 +1,38 @@
+import Nimble
+import RxSwift
+import RxTest
+
+/// A Nimble matcher that succeeds when the actual events emit an error
+/// of any type.
+public func throwError<T: Equatable>() -> Predicate<RecordedEvents<T>> {
+    func extractError(_ recorded: RecordedEvents<T>?) -> [Error]? {
```

Comment:
I could avoid having this extracted to a function, but I wanted to isolate swift version checks (`flatMap` vs. `compactMap`) without repeating myself too much.